File: sample/src/main/java/com/orhanobut/hawk/sample/MainActivity.java
package com.orhanobut.hawk.sample;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

import com.orhanobut.hawk.Hawk;

import java.io.Serializable;

public class MainActivity extends Activity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Hawk.init(this);
        Hawk.clear();

        test();
        test2();
    }

    void test() {
        Integer a = 5;
        Hawk.put(TAG, a);

        int b = Hawk.get(TAG);
        Log.d(TAG, String.valueOf(b));
    }

    void test2() {
        Foo foo = new Foo();
        Hawk.put("test2", foo);
        Foo foo1 = Hawk.get("test2");
    }

    static class Foo implements Serializable {
        String a = "foo";
    }

}

File: dagger-sample/src/main/kotlin/com/orhanobut/hawk/dagger/sample/MainActivity.kt
package com.orhanobut.hawk.dagger.sample

import android.os.Bundle
import android.support.annotation.VisibleForTesting
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import com.orhanobut.hawk.dagger.Hawker
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector
import javax.inject.Inject

@VisibleForTesting
class MainActivity : AppCompatActivity(), HasSupportFragmentInjector {

    @Inject
    lateinit var hawker: Hawker

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>

    override fun supportFragmentInjector() = dispatchingAndroidInjector

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        save()
        load()
    }

    private fun save() {
        hawker.put("name", "Hawker")
        hawker.put("description", "Hawk with Dagger 2 support")
    }

    private fun load() {
        val name: String? = hawker.get("name")
        val description: String? = hawker.get("description")
    }
}
